jbmc/src/java_bytecode/ci_lazy_methods.cpp

```diff
@@ -407,28 +397,35 @@ void ci_lazy_methodst::initialize_instantiated_classes(
   // As in class_loader, ensure these classes stay available
   for(const auto &id : extra_instantiated_classes)
     needed_lazy_methods.add_needed_class("java::" + id2string(id));
+
+  // Special case for enums. We may want to generalise this, see TG-4689
+  // and the comment in java_object_factoryt::gen_nondet_pointer_init.
+  for(const auto &found_class : needed_lazy_methods.get_instantiated_classes())
+  {
+    const auto &class_type = to_java_class_type(ns.lookup(found_class).type);
+    if(class_type.get_base("java::java.lang.Enum"))
```

Comment:
I think we can test on `ID_enumeration` for class types instead of this comparison.

Reply:
Huh, I've been doing this all over the place and no one told me about `ID_enumeration`. :confused: I'd like to get this merged quickly to fix my previous mistake, happy if I change all the `get_base`s in a follow-up PR?